notify/impl.go

```diff
@@ -108,8 +108,8 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 	for _, nc := range confs {
```

Comment:
When the above filtering is extracted, we can also process the logic of the loop below within createStage, I think, and don't have to build this tree out-of-band.

Might be overlooking something, though.